shortfin/python/shortfin_apps/llm/components/decoder/decoder.py

```diff
@@ -384,13 +377,29 @@ async def run(self, input_ids):
                 extra_token_slots=0,  # prefill needs no extra kvcache slots to write to
             )
         except CacheAllocationFailure:
+            logger.debug("Cannot fulfill request for %d pages", needed_pages)
             raise RuntimeError(
-                "Failed to allocate %d pages for prefill request. ", needed_pages
+                f"Failed to allocate {needed_pages} pages for prefill request."
             )
+
         logger.debug(f"Successfully acquired allocation: {allocation}")
         prefill_req.free_cache_pages()
         prefill_req.allocation = allocation
 
+    def _allocate_decode_cache(self, request: LlmInferenceExecRequest):
+        if request.allocation is not None:
+            request.allocation.extend_allocation(
```

Comment:
We should not be using the extend allocation command. This should ultimately be removed from the allocator